src/jdk.compiler/share/classes/com/sun/tools/javac/launcher/MemoryClassLoader.java

```diff
@@ -233,7 +229,7 @@ protected URL findResource(String moduleName, String name) throws IOException {
     public URL findResource(String name) {
         String binaryName = toBinaryName(name);
         if (binaryName == null || sourceFileClasses.get(binaryName) == null) {
-            return null;
+            return toResourceInRootPath(name); // can be null
```

Comment:
"can be null" is a bit confusing here. I initially read it that name can be null but this isn't possible, at least not with a well behaved class loader as the getResource methods are specified to reject null so will never delegate a null to findResource. The comment is really that it might return null (meaning not found).

Reply:
Yes. I can either update the comment to read `returning null here is expected by the caller` or refactor the method to read `toResourceInRootPathOrNull(String name)`. Another option would be to make the return type an `Optional<URL>` and handle the not-available case at the call site.

Reply:
I removed the comment without touching any names nor return types.